ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSerdeFactory.java

```diff
@@ -67,20 +81,22 @@ <T> Serde<T> createSerde(
       final boolean isKey) {
     validate(schema);
 
-    final Optional<Schema> physicalSchema = properties.getSchemaId().isPresent() ? Optional.of(
-        SerdeUtils.getAndTranslateSchema(srFactory, properties.getSchemaId()
-            .get(), new ProtobufSchemaTranslator(properties))) : Optional.empty();
+    final Optional<Schema> physicalSchema = getSRSchema(srFactory);
+
+    final ConnectSchema compatibleSchema = getSRSchema(srFactory)
```

Comment:
Why do we need to call `getSRSchema` twice here?

Reply:
My mistake, I removed it.